google/cloud/storage/_helpers.py

```diff
@@ -312,3 +402,44 @@ def _convert_to_timestamp(value):
     utc_naive = value.replace(tzinfo=None) - value.utcoffset()
     mtime = (utc_naive - datetime(1970, 1, 1)).total_seconds()
     return mtime
+
+
+def _add_generation_match_parameters(name_value_pairs, **parameters):
+    """Add generation match parameters into the given parameters list.
+
+    :type name_value_pairs: list or dict
+    :param name_value_pairs: Parameters list or dict.
+
+    :type parameters: dict
+    :param parameters: if*generation*match parameters to add.
+    """
+    for snakecase_name, camelcase_name in _GENERATION_MATCH_PARAMETERS:
+        value = parameters.get(snakecase_name)
+
+        if value is not None:
+            if isinstance(name_value_pairs, list):
+                name_value_pairs.append((camelcase_name, value))
+
+            elif isinstance(name_value_pairs, dict):
+                name_value_pairs[camelcase_name] = value
+
+
+def _raise_for_more_than_one_none(**kwargs):
+    """Raise ``ValueError`` exception if more than one parameter was set.
+
+    :type error: :exc:`ValueError`
+    :param error: Description of which fields were set
+
+    :raises: :class:`~ValueError` containing the fields that were set
+    """
+    if sum(arg is not None for arg in kwargs.values()) > 1:
+        escaped_keys = ["'%s'" % name for name in kwargs.keys()]
+
+        keys_but_last = ", ".join(escaped_keys[:-1])
+        last_key = escaped_keys[-1]
+
+        msg = "Pass at most one of {keys_but_last} and {last_key}".format(
+            keys_but_last=keys_but_last, last_key=last_key
+        )
+
+        raise ValueError(msg)
```

Comment:
Moved into `_helpers`. Seems to me the name of the helper is not very correct. It raises an error in case if more than one arg is **not** `None`. Maybe should be renamed into `_raise_for_more_than_one_set()`?

Reply:
Good catch. I agree we should change the name. If you prefer not to change the name in this PR, please open an issue and either tackle it separately or assign it to me so it doesn't get lost.